packages/snap/src/index.ts

```diff
@@ -89,103 +86,95 @@ interface RPCRequest {
   request: ShapeShiftSnapRPCRequest
 }
 
-export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
+export const onRpcRequest: OnRpcRequestHandler = async ({ request, origin }: RPCRequest) => {
   const { method, params } = request
   switch (method) {
     case 'avax_getAddress':
       return await avalancheGetAddress(params)
     case 'avax_signMessage':
-      return await avalancheSignMessage(params)
-    case 'avax_signTransaction':
```

Comment:
Do we want to remove all these methods for EVM chains things, since we don't support 60th slip44 and EVM things are passing through MM as usual, without any snap involved?

Reply:
No, we want to leave them in place. Doing the EVM stuff in the hdwallet adapter is just a temporary hack to get us around the upstream bug with `wallet_switchEthereumChain`. Untilately, the snap should call those methods internally using the Dapp API.